presto-common/src/main/java/com/facebook/presto/common/type/TimestampType.java

```diff
@@ -16,21 +16,41 @@
 import com.facebook.presto.common.block.Block;
 import com.facebook.presto.common.function.SqlFunctionProperties;
 
+import java.util.Objects;
+import java.util.concurrent.TimeUnit;
+
 import static com.facebook.presto.common.type.TypeSignature.parseTypeSignature;
+import static java.util.concurrent.TimeUnit.MICROSECONDS;
 import static java.util.concurrent.TimeUnit.MILLISECONDS;
 
 //
-// A timestamp is stored as milliseconds from 1970-01-01T00:00:00 UTC.  When performing calculations
+// TIMESTAMP is stored as milliseconds from 1970-01-01T00:00:00 UTC.  When performing calculations
+// on a timestamp the client's time zone must be taken into account.
+// TIMESTAMP_MICROSECONDS is stored as microseconds from 1970-01-01T00:00:00 UTC.  When performing calculations
 // on a timestamp the client's time zone must be taken into account.
 //
 public final class TimestampType
         extends AbstractLongType
 {
-    public static final TimestampType TIMESTAMP = new TimestampType();
+    private final TimeUnit precision;
+    public static final TimestampType TIMESTAMP = new TimestampType(MILLISECONDS);
+    public static final TimestampType TIMESTAMP_MICROSECONDS = new TimestampType(MICROSECONDS);
 
-    private TimestampType()
+    private static String getType(TimeUnit precision)
```

Comment:
do we need `static` here ?

Reply:
I generally prefer to mark the methods as static, when they are just utility helper functions. What would you recommend ?

Reply:
yeah okay makes sense.